src/libs/data-table-provenance.js

```diff
@@ -88,3 +90,98 @@ export const useColumnProvenance = (workspace, entityType) => {
     loadColumnProvenance: loading ? _.noop : () => { loadColumnProvenance() }
   }
 }
+
+export const fileProvenanceTypes = Utils.enumify(['externalFile', 'unknown', 'maybeSubmission', 'workflowOutput', 'workflowLog'])
+
+const isLog = (url, task) => _.some(log => _.includes(url, [log.backendLogs.log, log.stdout, log.stderr]), task.logs)
+
+const isOutput = (url, task) => _.some(
+  // outputs can contain a value, a collection of values, or a collection of collections of values
+  output => output === url || _.includes(url, output) || _.some(_.includes(url), output),
+  task.outputs
+)
+
+const getFileProvenance = async (workspace, fileUrl, { signal } = {}) => {
```

Comment:
You could consider unit testing this as well.

Reply:
Added tests in https://github.com/DataBiosphere/terra-ui/pull/3353/commits/51d53a2fa19e9e46f8471c67cb7a99a4b5ab8f4e.